internal/scout/usage/k8s.go

```diff
@@ -0,0 +1,384 @@
+package usage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/charmbracelet/bubbles/table"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/src-cli/internal/scout/style"
+
+	"gopkg.in/inf.v0"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	metav1beta1 "k8s.io/metrics/pkg/apis/metrics/v1beta1"
+	metricsv "k8s.io/metrics/pkg/client/clientset/versioned"
+)
+
+type ContainerMetrics struct {
+	pod    string
+	limits map[string]Resources
+}
+
+type Resources struct {
+	cpu     *resource.Quantity
+	memory  *resource.Quantity
+	storage *resource.Quantity
+}
+
+const (
+	ABillion float64 = 1000000000
+)
+
+func K8s(
+	ctx context.Context,
+	clientSet *kubernetes.Clientset,
+	metricsClient *metricsv.Clientset,
+	rest *rest.Config,
+	opts ...Option,
+) error {
+	cfg := &Config{
+		namespace:     "default",
+		docker:        false,
+		pod:           "",
+		container:     "",
+		spy:           false,
+		k8sClient:     clientSet,
+		dockerClient:  nil,
+		metricsClient: metricsClient,
+	}
+
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	return renderUsageTable(ctx, cfg)
+}
+
+// renderUsageTable renders a table displaying resource usage for pods.
+// It accepts:
+// - ctx: The context for the request
+// - cfg: A config struct containing:
+//   - k8sClient: A Kubernetes clientset for accessing the API server
+//   - namespace: The namespace of the pods
+//
+// It returns:
+// - Any error that occurred while rendering the table
+func renderUsageTable(ctx context.Context, cfg *Config) error {
+	podInterface := cfg.k8sClient.CoreV1().Pods(cfg.namespace)
+	podList, err := podInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error listing pods: ")
```

Comment:
Nit: At this point, we know it's an error. So we don't really need to list out `error` in the error message itself. Otherwise when looking over log output we get messages like "error: error listing pods: error no such file found"... Something like "could not list pods: \<wrapped error\>" would be more concise. Same for all these messages.